### DISCUSSION TITLE
Basic question about how PUDL maps EIA data to its schema

### DISCUSSION THREAD

**Discussion by jmg80525** [catalyst-cooperative/pudl#1682] [Help Me!] [answered]
Investigating PUDL.  Took time to query some  data around the entity for whom  I work, PRPA which operates Rawhide (6761).  I queried some data on plant_id_eia (aka PlantCode) 6761 from the PUDL [generation_eia923 table](https://data.catalyst.coop/pudl/generation_eia923) and was perplexed to find in didn't initially agree with the the results I pulled from the [EIA v2 API](https://www.eia.gov/opendata/browser/electricity/facility-fuel).  Further investigation revealed that PUDL had not included generation from the[ NG turbines at the plant](https://www.prpa.org/generation/).  I found this odd since the PUDL metadata indicates the data is associated EIA Form 923 Schedule 3 and that same form contains entry in Part B of Schedule 3 page 7 [FUEL AND GENERATION INFORMATION FOR SINGLE CYCLE GAS TURBINES](https://www.eia.gov/survey/form/eia_923/proposed/2023/form.pdf). Please inform my as to why PUDL choose to associate only coal-fire steam turbines generation for data in the above mentioned table or what I am missing in my initial investigation. Cheers.

**Comment by zaneselvans** [comment, marked as answer]
There are several different places that "generation" is reported in the EIA 923, and IIRC, only around half of all generation gets reported to the `generation_eia923` table on a per-generator basis.  However, *all* generation *does* show up in the `generation_fuel_eia923` table, on the basis of plant ID, prime mover, and fuel. This is admittedly a weird combo, and it does not map directly to individual generators, but that's how the data is reported. I took a quick look at the EIA v2 API and it looked like they were aggregating all the generation and fuel consumption at the plant ID level, right?

Querying [generation_fuel_eia923 for plant ID 6761](https://data.catalyst.coop/pudl/generation_fuel_eia923?plant_id_eia__exact=6761) you can see a wider range of generation, including a bunch of natural gas fired gas turbines. I suspect if you aggregate the net generation by plant ID in that table, you should get numbers which are very similar to what you'd find at the plant level in the EIA API.  [Here's that aggregation](https://data.catalyst.coop/pudl?sql=SELECT%0D%0A++plant_id_eia%2C%0D%0A++report_date%2C%0D%0A++SUM%28fuel_consumed_mmbtu%29+AS+fuel_consumed_mmbtu%2C%0D%0A++SUM%28net_generation_mwh%29+AS+net_generation_mwh%0D%0AFROM%0D%0A++generation_fuel_eia923%0D%0AWHERE%0D%0A++plant_id_eia+%3D%3D+6761%0D%0AGROUP+BY%0D%0A++plant_id_eia%2C%0D%0A++report_date%3B) if you want to compare and let us know!

The availability of more granular generator or prime-mover level data is one of the main reasons that we go to all of the trouble of working with the spreadsheets EIA publishes, rather than using the API. Being able to pull out the operating costs and emissions of individual generators is necessary for regulatory proceedings where one unit may have very different characteristics from another.

We've got some routines which try to attribute the generation and fuel consumption that's reported by plant-primemover-fuel in the `generation_fuel_eia923` table to individual generators or generation units, but it's not perfect.

**Reply by jmg80525** [reply]
Thanks for the prompt and thorough response.  I've got another question that I'll spin up as another thread.
